akka-projection-jdbc/src/test/scala/akka/projection/jdbc/internal/JdbcSettingsSpec.scala

```diff
@@ -0,0 +1,201 @@
+      }
+    }
+
+    "throw exception when dialect not defined" in {
+
+      val config: Config =
+        ConfigFactory.parseString("""
+          akka {
+            loglevel = "DEBUG"
+            projection.jdbc = {
+              blocking-jdbc-dispatcher.thread-pool-executor.fixed-pool-size = 5
+            }
+          }
+          """)
+
+      createSystem(config) { sys =>
+        val msg =
+          intercept[IllegalArgumentException] {
+            JdbcSettings(sys)
+          }.getMessage
+
+        msg shouldBe "Dialect type not set. Settings 'akka.projection.jdbc.dialect' currently set to []"
+      }
+    }
+
+    "throw exception when dialect is unknown" in {
+
+      val config: Config =
+        ConfigFactory.parseString("""
+          akka {
+            loglevel = "DEBUG"
+            projection.jdbc = {
+              dialect = "h3-dialect"
+              blocking-jdbc-dispatcher.thread-pool-executor.fixed-pool-size = 5
+            }
+          }
+          """)
+
+      createSystem(config) { sys =>
+        val msg =
+          intercept[IllegalArgumentException] {
+            JdbcSettings(sys)
+          }.getMessage
+
+        msg shouldBe "Unknown dialect type: [h3-dialect]. Check settings 'akka.projection.jdbc.dialect'"
+      }
+    }
+
+    "throw exception when pool size not defined" in {
+
+      val config: Config =
+        ConfigFactory.parseString("""
+          akka {
+            loglevel = "DEBUG"
+            projection.jdbc = {
+              dialect = "h2-dialect"
+            }
+          }
+          """)
+
+      createSystem(config) { sys =>
+        intercept[IllegalArgumentException] {
+          JdbcSettings(sys)
+        }
+      }
+    }
+
+    "dont' fail if user decide to step out of a thread-pool-executor" in {
+      val config: Config =
+        ConfigFactory.parseString("""
+          akka {
+            loglevel = "DEBUG"
+            projection.jdbc = {
+              dialect = "h2-dialect"
+              
+              # user tweaks the dispatcher to be something else
+              # we can't check the pool size if it's not a thread-pool-executor
+              blocking-jdbc-dispatcher {
+                executor = "fork-join-executor"
+                fork-join-executor {
+                  parallelism-min = 2
+                  parallelism-factor = 2.0
+                  parallelism-max = 10
+                }
+                throughput = 1
+              }
+            }
+          }
+          """)
+
+      createSystem(config) { sys =>
+        JdbcSettings(sys)
+      }
+
+    }
+  }
+
+  private def createSystem(config: Config)(func: ActorSystem[Any] => Unit) = {
+    val sys = ActorSystem(Behaviors.empty[Any], "test-sys", config)
+    func(sys)
+    sys.terminate()
+    Await.ready(sys.whenTerminated, 3.seconds)
```

Comment:
```suggestion
    func(sys)
    Await.ready(sys.terminate(), 3.seconds)
```

Nitpick.

Reply:
`sys.terminate()` already returns a future that will complete when the termination is (urgh) _completed_.

Reply:
I don't think so.

```scala
/**
   * Terminates this actor system by running [[akka.actor.CoordinatedShutdown]] with reason
   * [[akka.actor.CoordinatedShutdown.ActorSystemTerminateReason]].
   *
   * If `akka.coordinated-shutdown.run-by-actor-system-terminate` is configured to `off`
   * it will not run `CoordinatedShutdown`, but the `ActorSystem` and its actors
   * will still be terminated.
   *
   * This will stop the guardian actor, which in turn
   * will recursively stop all its child actors, and finally the system guardian
   * (below which the logging actors reside).
   *
   * This is an asynchronous operation and completion of the termination can
   * be observed with [[ActorSystem.whenTerminated]] or [[ActorSystem.getWhenTerminated]].
   */
  def terminate(): Unit
```

Reply:
That's true for the classic ActorSystem, we are using the typed one.

Reply:
> That's true for the classic ActorSystem, we are using the typed one.

Ooooh! TIL